tests/test_tasks.py

```diff
@@ -163,6 +164,7 @@ def my_task():
 
 
 class TestTaskCall:
+    @pytest.mark.skipif(PREFECT_VERSION < "2.15.0", reason="now allowed")
```

Comment:
perhaps we should an experimental setting `PREFECT_ALLOW_TASK_AUTONOMY`? or something